src/vaadin-tabs-mixin.ts

```diff
@@ -0,0 +1,139 @@
+import { SelectionInViewClass } from '@vaadin/selection-in-view-mixin/selection-in-view-class.js';
+import {
+  SingleSelectionMixin,
+  SingleSelectionInterface
+} from '@vaadin/single-selection-mixin/single-selection-mixin.js';
+import { SlottedItemsMixin, SlottedItemsInterface } from '@vaadin/slotted-items-mixin/slotted-items-mixin.js';
+import { ResizableMixin } from '@vaadin/resizable-mixin/resizable-mixin.js';
+import { ResizableClass } from '@vaadin/resizable-mixin/resizable-class.js';
+import { tabsStyles } from './vaadin-tabs-css';
+
+type TabsBase = new () => LitElement;
+
+type Tab = new () => LitElement &
+  SlottedItemsInterface &
+  KeyboardDirectionInterface &
+  OrientationInterface &
+  ResizableClass &
+  SingleSelectionInterface &
+  SelectionInViewClass;
+
+export const TabsMixin = <T extends TabsBase>(base: T): Tab => {
+  class Tabs extends ResizableMixin(
+    SelectionInViewMixin(
+      SingleSelectionMixin(OrientationMixin(RovingTabIndexMixin(KeyboardDirectionMixin(SlottedItemsMixin(base)))))
+    )
+  ) {
+    private _scroller?: HTMLElement;
+
+    constructor() {
+      super();
+
+      this.selected = 0;
+    }
+
+    static get styles() {
+      return tabsStyles;
+    }
+
+    protected render() {
+      return html`
+        <div @click="${this._scrollBack}" part="back-button"></div>
+
+        <div part="tabs" @scroll="${this._onScroll}">
+          <slot></slot>
+        </div>
+
+        <div @click="${this._scrollForward}" part="forward-button"></div>
+      `;
+    }
+
+    protected firstUpdated(props: PropertyValues) {
+      super.firstUpdated(props);
+
+      this.setAttribute('role', 'tablist');
+      this._scroller = this.renderRoot.querySelector('[part="tabs"]') as HTMLElement;
+
+      // Wait for the vaadin-tab elements to upgrade and get styled
+      const tabsComplete = this.items.map(tab => (tab as LitElement).updateComplete);
+      Promise.all(tabsComplete).then(() => {
+        this._updateOverflow();
+      });
+    }
+
+    protected updated(props: PropertyValues) {
+      if (props.has('orientation') || props.has('_items')) {
+        this.items.forEach(item => {
+          if (this.orientation) {
+            item.setAttribute('orientation', this.orientation);
+          } else {
+            item.removeAttribute('orientation');
+          }
+        });
+
+        this._updateOverflow();
+      }
+
+      // Ensure scroll to item is called after "overflow" update
+      super.updated(props);
+    }
+
+    protected _sizeChanged(contentRect: DOMRect) {
+      this._updateOverflow();
+
+      super._sizeChanged && super._sizeChanged(contentRect);
+    }
+
+    private _onScroll() {
+      this._updateOverflow();
+    }
+
+    private get _overflow() {
+      const { scrollTop, scrollLeft, scrollHeight, scrollWidth } = this._scrollTarget;
+      const scrollPosition = this._vertical ? scrollTop : scrollLeft;
```

Comment:
Question: if we are going to use `isVertical` from `KeyboardDirection` and we anyway need to override it in tabs. Can we at the same time get rid out of `_vertical` in `SelectionInView` and use `isVertical` from `KeyboardDirection` over there and here as well? 🤔

Reply:
Now we have `_vertical` used in all mixins, so no need to change it here.